pkg/controller/cloneset/core/cloneset_core.go

```diff
@@ -201,14 +203,25 @@ func (c *commonControl) IgnorePodUpdateEvent(oldPod, curPod *v1.Pod) bool {
 		}
 		return false
 	}
+	isPodInplaceUpdating := func(pod *v1.Pod) bool {
+		if len(pod.Labels) > 0 && appspub.LifecycleStateType(pod.Labels[appspub.LifecycleStateKey]) != appspub.LifecycleStateNormal {
+			return true
+		}
+		return false
+	}
 
-	if containsReadinessGate(curPod) {
+	if containsReadinessGate(curPod) || isPodInplaceUpdating(curPod) {
 		opts := c.GetUpdateOptions()
 		opts = inplaceupdate.SetOptionsDefaults(opts)
 		if err := containersUpdateCompleted(curPod, opts.CheckContainersUpdateCompleted); err == nil {
 			if cond := inplaceupdate.GetCondition(curPod); cond == nil || cond.Status != v1.ConditionTrue {
 				return false
 			}
+			// if InPlaceWorkloadVerticalScaling is enabled, we should not ignore the update event of updating pod
+			// for handling only in-place resource resize
+			if utilfeature.DefaultFeatureGate.Enabled(features.InPlaceWorkloadVerticalScaling) {
+				return false
```

Comment:
shall we compare the pod.status.resize between old and new object, and ignore the event if the resize is the same ?

also, this func is not covered by UT, plz add an ut

Reply:
When we only resize limit, the resize is always nil

Reply:
or just compare the status.resources between old and new object ?